### PR TITLE
STIG update RHEL7 network_disable_ddns_interfaces

### PR DESCRIPTION
**Pull request by redhatrises**
- Add XCCDF and OVAL content for disabling DDNS for clients

### CODE REVIEW

File: RHEL/7/input/xccdf/system/network/network.xml [hunk 1]
```diff
@@ -31,6 +31,34 @@ communication over the loopback device, then disable this service.
...
+<pre>$ grep -rni "send host-name" /etc/dhclient.conf /etc/dhcp</pre>
+The output should return no results.
+</ocil>
+<rationale>
+Dynamic DNS updates transmit unencrypted information about a system
+including its name and address and should not be used unless needed.
+</rationale>
+<ident cce="RHEL7-CCE-TBD" />
+<oval id="network_disable_ddns_interfaces" />
+<ref nist="CM-6(b)" disa="366" ossrg="SRG-OS-000480-GPOS-00227" stigid="040840" />
```
Comment (shawndwells) [inline RHEL/7/input/xccdf/system/network/network.xml]: 040840 is for PEERDNS settings:
image


I can't find this check from DISA -- I must be totally blind today =/

Comment (redhatrises) [inline RHEL/7/input/xccdf/system/network/network.xml]: @shawndwells the check and fix about `PEERDNS` is what is wrong. `PEERDNS` updates `resolv.conf` on the client. However, the requirement and vulnerability discussion are about the client sending unencrypted Dynamic DNS updates which is done with `DHCP_HOSTNAME` or `send host-name hostname.fqdn` with `dhclient.conf`

However if it truly is about `PEERDNS` then the requirement and vulnerability discussion are wrong.

**Comment by iankko** [comment]
@shawndwells @redhatrises 
If I got the current status on this one right, it's stuck because the rule's title doesn't correspond / match to the instructions / steps, which should be actually performed, right? IOW we are not sure if the requirement wants dynamic DNS configuration (which for RHEL-7 seems to be achieved by using sssd: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Windows_Integration_Guide/sssd-ad-options.html#sssd-dyndns ). For PEERDNS for RHEL-6 it truly seems to be achieved by proper network interface configuration (https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Deployment_Guide/s1-networkscripts-interfaces.html )

Is it possible to clarify what's necessary to be performed to move with this one? (should we rewrite the OVAL as provided now or is there something to be confirmed by DISA committee)?

**Comment by redhatrises** [comment]
@iankko correct.

The rule status that "the DHCP client must not send any updates". However,

1. PEERDNS gets dns servers and updates /etc/resolv.conf but does not "send" DNS updates to the server. This is a pull.
2.  SSSD ability to implement Dynamic Updates is for AD only.
3. Setting DHCP_HOSTNAME and/or configuration in dhclient.conf will "send" updates to the DNS servers.

Of course if I am wrong about the above, feel free to chime in.

**Comment by iankko** [comment]
@redhatrises 
Can you fix the conflicts on this one?

**Comment by redhatrises** [comment]
@iankko fixed merge conflict.

**Comment by iankko** [comment]
@redhatrises 
> @iankko fixed merge conflict.

Great. Thanks!

LGTM. ACK (should there be issues reported by the users we will fix them in the future).

**Pull request merged**